libraries/networking/src/udt/NetworkSocket.cpp

```diff
@@ -33,6 +33,18 @@ NetworkSocket::NetworkSocket(QObject* parent) :
 #endif
 }
 
+NetworkSocket::~NetworkSocket() {
+    disconnect(&_udpSocket, &QUdpSocket::readyRead, this, &NetworkSocket::readyRead);
+    disconnect(&_udpSocket, &QAbstractSocket::stateChanged, this, &NetworkSocket::onUDPStateChanged);
+    disconnect(&_udpSocket, SIGNAL(error(QAbstractSocket::SocketError)),
+        this, SLOT(onUDPSocketError(QAbstractSocket::SocketError)));
+
+#if defined(WEBRTC_DATA_CHANNELS)
+    disconnect(&_webrtcSocket, &WebRTCSocket::readyRead, this, &NetworkSocket::readyRead);
+    disconnect(&_webrtcSocket, &WebRTCSocket::stateChanged, this, &NetworkSocket::onWebRTCStateChanged);
+    // WEBRTC TODO: Add similar for errorOccurred
```

Comment:
Do this now or defer because ...?

Reply:
There is a similar TODO in the constructor where these connections are made, so no connection yet to disconnect, just duplicated the comment since it would apply here too.